src/utils/validators.js

```diff
@@ -0,0 +1,305 @@
+ *
+ * @throws {TypeError} Value should be a valid int64 amount.
+ */
+export function validateAmount ({
+  value,
+  fieldName = '',
+  allowZero,
+  max,
+  min,
+  maxDecimalPlaces
+}) {
+  if (!BaseOperation.isValidAmount(value, allowZero, max, min, maxDecimalPlaces)) {
+    throw new TypeError(
+      `${fieldName} must be a valid int64 string with params:` +
+      `${JSON.stringify({ min, max, allowZero, maxDecimalPlaces })}, ` +
+      `got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/**
+ * Validate value to be a double string with specified precision.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ * @param {number} [opts.min] The minimum amount.
+ * @param {number} [opts.max] The maximum amount.
+ * @param {number} [opts.maxDecimalPlaces] The maximum decimal places allowed.
+ *
+ * @throws {TypeError} Value should be a valid double string with specified precision.
+ */
+export function validateDouble ({
+  value,
+  fieldName = '',
+  max,
+  min,
+  maxDecimalPlaces
+}) {
+  validateAmount({
+    value,
+    fieldName,
+    min,
+    max,
+    maxDecimalPlaces,
+    allowZero: true
+  })
+}
+
+/**
+ * Validate value to be a uint64 string.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ * @param {number} [opts.min] The minimum amount.
+ * @param {number} [opts.max] The maximum amount.
+ *
+ * @throws {TypeError} Value should be a valid uint64 string.
+ */
+export function validateUint64 ({
+  value,
+  fieldName = '',
+  max,
+  min
+}) {
+  validateAmount({
+    value,
+    fieldName,
+    min,
+    max,
+    maxDecimalPlaces: 0,
+    allowZero: true
+  })
+}
+
+/**
+ * Validate value to be an asset code.
+ *
+ * @param {object} opts
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid asset code.
+ */
+export function validateAssetCode ({ value, fieldName = '' }) {
+  if (!BaseOperation.isValidAsset(value)) {
+    throw new TypeError(
+      `${fieldName} must be valid asset code (alphanumeric string` +
+      `with the length of 1-16, got ${JSON.stringify(value)}`
+    )
+  }
+}
+
+/**
+ * Validate value to be a XDR-specified enum type instance.
+ *
+ * @param {object} opts
+ * @param {xdr.*} opts.type The XDR enum type to validate.
+ * @param {*} opts.value The value to validate.
+ * @param {string} opts.fieldName The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a XDR-specified enum type instance.
+ */
+export function validateXdrEnumType ({ value, type, fieldName = '' }) {
+  if (!(value instanceof type)) {
+    throw new TypeError(
+      `${fieldName} must be a valid xdr.${type.enumName}, got ${JSON.stringify(value)}`
```

Comment:
since we got here type, let’s also throw an error if no `enumName` found in `enum` to prevent things like "cannot find enumName of undefined"

Reply:
Updated